pkg/engine/lifecycletest/pulumi_test.go

```diff
@@ -311,19 +311,6 @@ func TestLanguageHostDiagnostics(t *testing.T) {
 				evts []Event, res result.Result) result.Result {
 
 				assertIsErrorOrBailResult(t, res)
-				sawExitCode := false
```

Comment:
Why is this change needed? If I add it back the test still passes. 

AFAIK this test is testing what we call the engine (pkg/engine) and is unrelated to the Python SDK. It's trying to test that the engine would propagate the text of a returned error from the user program into a DiagEvent with:

```
sawExitCode = strings.Contains(msg, errorText) && e.Severity == diag.Error
```

It still should be the case no?

Reply:
This is a remnant from splitting out the Golang PR. I'll add it back.

Reply:
Thanks! That's the only part that looks odd. The rest looks great!